### PR TITLE
docs: dialog written docs PTL-1590

### PR DESCRIPTION
**Pull request by matteematt**

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+- overlay
+- popup
+sidebar_position: 8
+---
+
+import DialogDemo from '/examples/ui/client-capabilities/interaction/dialog.js';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Anchors add different types of links to your page.
```
Comment (Gareth-Spencer-Genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx:20]: Copy and paste error?

File: docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+      }
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+
+## API
+
+### Attributes
```
Comment (Gareth-Spencer-Genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx:145]: I think some attributes are missing, looking in foundation-ui dialog.ts there are attributes like show-close-icon etc.

File: docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+|Method|Description|Example|
+|--|--|--|
+|show|Show the dialog|`dialogRef.show()`|
+|hide|Hides the dialog|`dialogRef.hide()`|
+
+### Slots
+
+|Slot|Description|
+|---|---|
+|Default|All the content which you want to display on the dialog box when shown.|
```
Comment (Gareth-Spencer-Genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx:247]: I think some slots are missing, looking in foundation-ui dialog.ts there are slots like "top".

**Comment by matteematt** [comment]
@Gareth-Spencer-Genesis just realising that our Dialog component isn't the FAST component like everything else but something we seem to have written from scratch. Need to look into that, putting back in draft for now

File: docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx (hunk 1 above)
Comment (Gareth-Spencer-Genesis) [reply docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx:20]: This line is still there on the latest:
"Anchors add different types of links to your page"

File: docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    ```
+  </TabItem>
+</Tabs>
+
+
+## API
+
+### Attributes
+
+|Attribute|Type|Description|Example|
```
Comment (Gareth-Spencer-Genesis) [inline docs/001_develop/03_client-capabilities/008_interaction/dialog.mdx:147]: these tables should use the <table> <tr> <td> structure

**Pull request merged**